internal/sync/sync.go

```diff
@@ -261,6 +263,47 @@ const (
 	maxAcceptableDispatch = time.Millisecond * 50
 )
 
+// Returns remote protocol version as a string.
+// Or, an error if one exists.
+func (s Sync) Version() (string, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	defer cancel()
+
+	conn, err := s.Client.DialWsep(ctx, s.Env)
+	if err != nil {
+		return "", err
+	}
+	defer conn.Close(websocket.CloseNormalClosure, "")
+
+	execer := wsep.RemoteExecer(conn)
+	process, err := execer.Start(ctx, wsep.Command{
+		Command: "rsync",
+		Args:    []string{"--version"},
+	})
+	if err != nil {
+		return "", err
+	}
+	buf := &bytes.Buffer{}
+	go io.Copy(buf, process.Stdout())
```

Comment:
I would remove the `go` here. `io.Copy` will will terminate as expected once the command completes.

Having a goroutine here can be racy, because there's no guarantees that it runs or completes the copy before we start reading from the buffer.

Reply:
Done.